### PR TITLE
[Security] [Guard] Remove unnecessary parameter

### PR DESCRIPTION
**Pull request by Arman-Hosseini**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| License       | MIT

$providerKey parameter is provided in the  body of the method.

### CODE REVIEW

File: src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function __construct(TokenStorageInterface $tokenStorage, EventDispatcher
     /**
      * Authenticates the given token in the system.
      */
-    public function authenticateWithToken(TokenInterface $token, Request $request/*, string $providerKey */)
+    public function authenticateWithToken(TokenInterface $token, Request $request)
     {
         $providerKey = \func_num_args() > 2 ? func_get_arg(2) : null;
```
Comment (chalasr) [inline src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: The argument is used, we should rather uncomment it on 5.x (with null default value to preserve the current behavior).

Comment (Arman-Hosseini) [reply src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: So we have to close this request?

Comment (chalasr) [reply src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: Or you can rebase your branch on the 5.x branch (and change the base branch of your PR accordingly), then uncomment the argument as I explained before.

Comment (Arman-Hosseini) [reply src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: Comment have already been removed since 4.2 branch.

Comment (derrabus) [reply src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: This comment is the usual way how we document a new parameter that is to be introduced with the next major release.

**Comment by derrabus** [comment]
Let‘s close the PR. Thanks for suggesting.

File: src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php (hunk 1 above)
Comment (Arman-Hosseini) [reply src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php:57]: Then what happens if this parameter is removed again in the next version? Will there be a comment again?

**Pull request closed without merging**